pyiceberg/table/__init__.py

```diff
@@ -621,6 +621,24 @@ def delete(self, delete_filter: Union[str, BooleanExpression], snapshot_properti
         if not delete_snapshot.files_affected and not delete_snapshot.rewrites_needed:
             warnings.warn("Delete operation did not match any records")
 
+    def has_duplicates(self, file_paths: List[str]) -> bool:
+        unique_files = set(file_paths)
+        return len(unique_files) != len(file_paths)
+
+    def find_referenced_files(self, file_paths: List[str]) -> list[str]:
```

Comment:
FYI there's also the `table.inspect.files()` API 
https://py.iceberg.apache.org/api/#files
which returns all data files in the current snapshot

Reply:
I guess we only need to worry about the current snapshot. 
If a data file existed in previous snapshots, but not in the current, we can still add the file.